src/pe/relocation.rs

```diff
@@ -131,3 +308,370 @@ impl<'a> Iterator for Relocations<'a> {
+    /// The starting RVA for the relocation block.
+    pub rva: u32,
+    /// The total size of the relocation block, in bytes.
+    pub size: u32,
+    /// The bytes of [`RelocationWord`] data within this block.
+    pub bytes: &'a [u8],
+}
+
+impl RelocationBlock<'_> {
+    /// Returns iterator for [`RelocationWord`]
+    pub fn words(&self) -> RelocationWordIterator {
+        RelocationWordIterator {
+            bytes: &self.bytes,
+            offset: 0,
+        }
+    }
+}
+
+/// Raw size of [`RelocationBlock`]
+pub const RELOCATION_BLOCK_SIZE: usize = 8;
+
+impl<'a> scroll::ctx::TryFromCtx<'a, scroll::Endian> for RelocationBlock<'a> {
+    type Error = crate::error::Error;
+
+    fn try_from_ctx(bytes: &'a [u8], ctx: scroll::Endian) -> Result<(Self, usize), Self::Error> {
+        let mut offset = 0;
+        let rva = bytes.gread_with::<u32>(&mut offset, ctx)?;
+        let size = bytes.gread_with::<u32>(&mut offset, ctx)?;
+        if (size as usize) < RELOCATION_BLOCK_SIZE {
+            return Err(error::Error::Malformed(format!(
+                "base reloc size {size} is less than a block {RELOCATION_BLOCK_SIZE}"
+            )));
+        }
+        let word_size = size as usize - RELOCATION_BLOCK_SIZE;
+        if offset + word_size > bytes.len() {
+            return Err(error::Error::Malformed(format!(
+                "base reloc block offset {:#x} and size {:#x} exceeds the bounds of the bytes size {:#x}",
+                offset,
+                word_size,
+                bytes.len()
+            )));
+        }
+        let bytes = &bytes[offset..offset + word_size];
+        offset += word_size;
+        Ok((Self { rva, size, bytes }, offset))
+    }
+}
+
+#[derive(Clone, Copy, Debug)]
+pub struct RelocationBlockIterator<'a> {
+    bytes: &'a [u8],
+    offset: usize,
+}
+
+impl<'a> Iterator for RelocationBlockIterator<'a> {
+    type Item = error::Result<RelocationBlock<'a>>;
+
+    fn next(&mut self) -> Option<Self::Item> {
+        if self.offset >= self.bytes.len() {
+            return None;
+        }
+
+        Some(
+            match self
+                .bytes
+                .gread_with::<RelocationBlock>(&mut self.offset, scroll::LE)
+            {
+                Ok(block) => Ok(block),
+                Err(error) => {
+                    self.bytes = &[];
+                    Err(error.into())
+                }
+            },
+        )
+    }
+}
+
+impl FusedIterator for RelocationBlockIterator<'_> {}
+
+/// Represents a single relocation entry within a PE base relocation block.
+///
+/// Each relocation entry is stored as a 16-bit value ([`u16`]), referred to as a "relocation word."
+/// The value is divided into two components:
+///
+/// * The upper 4 bits specify the relocation type, which determines how the relocation
+///   should be applied.
+/// * The remaining 12 bits specify the offset from the block's starting RVA where the
+///   relocation should be applied.
+#[repr(C)]
```

Comment:
Does this need to be repr(C)? Are you using it across FFI boundaries or constructing it and sending to C or vice versa? If not I’d opt to remove, and can add later if necessary since this is pub and it’ll be part of the pub api with that layout constraint

Reply:
I don't think this needs to be a `#[repr(C)]`, thanks for the suggestion!